frontend/src/pages/account/Security/dialogs/TokenDialog.vue

```diff
@@ -0,0 +1,145 @@
+                    Name
+                    <template #description>Token name</template>
+                </FormRow>
+                <ff-checkbox v-model="input.expires" data-form="expiry-toggle" label="Add Expiry Date" />
+                <FormRow v-model="input.expiresAt" data-form="token-expiry" type="date" :disabled="!input.expires">
+                    Expires
+                    <!-- <template v-slot:description>Expires</template> -->
+                </FormRow>
+            </form>
+        </template>
+    </ff-dialog>
+</template>
+
+<script>
+import settings from '../../../../api/settings.js'
+import userApi from '../../../../api/user.js'
+
+import FormRow from '../../../../components/FormRow.vue'
+
+export default {
+    name: 'TokenDialog',
+    components: {
+        FormRow
+    },
+    emits: ['token-created', 'token-updated'],
+    setup () {
+        return {
+            showCreate () {
+                this.$refs.dialog.show()
+                this.token = null
+                this.input = {
+                    id: null,
+                    name: '',
+                    expiresAt: null,
+                    expires: false
+                }
+                this.edit = false
+            },
+            showEdit (row) {
+                this.token = row
+                this.input = {
+                    id: row.id,
+                    name: row.name
+                }
+                if (row.expiresAt === null) {
+                    this.input.expires = false
+                } else {
+                    this.input.expires = true
+                    this.input.expiresAt = row.expiresAt.split('T')[0] // `${d.getFullYear()}-${d.getMonth()+1}-${d.getDate()}`
+                }
+                this.edit = true
+                this.$refs.dialog.show()
+            }
+        }
+    },
+    data () {
+        return {
+            token: null,
+            scopes: [],
+            input: {
+                id: null,
+                name: '',
+                scope: {},
+                expiresAt: null,
+                expires: true
+            },
+            edit: false
+        }
+    },
+    computed: {
+        dialogTitle () {
+            if (this.token) {
+                return 'Update token'
+            } else {
+                return 'Create token'
+            }
+        },
+        disableConfirm () {
+            if (!this.input.name) {
+                return true
+            }
+            if (!this.input.expires) {
+                return false
+            } else {
+                if (this.input.expiresAt) {
+                    return false
+                }
+            }
+            return true
+        }
+    },
+    mounted () {
+        this.getScopes()
+    },
+    methods: {
+        getScopes: async function () {
+            const temp = await settings.getSettings()
+            this.scopes = Object.keys(temp['platform:auth:permissions'])
+        },
```

Comment:
```suggestion
    methods: {
```
As per other comment - not including scopes in this iteration, so suggest removing this code as we may want a different approach in the future.